### DISCUSSION TITLE
DatabaseUUIDEncodingStrategy is not concurrency-safe in Swift 6?

### DISCUSSION THREAD

**Discussion by adamek314** [groue/GRDB.swift#1567] [General]
I'm trying out MacOS 15 Sequoia, Xcode 16 beta and Swift 6. I get a warning at the following code:

`extension Categories: TableRecord {
    static let databaseTableName: String = "categories"
    static let databaseUUIDEncodingStrategy = DatabaseUUIDEncodingStrategy.uppercaseString`

The warning is:
> Static property 'databaseUUIDEncodingStrategy' is not concurrency-safe because non-'Sendable' type 'DatabaseUUIDEncodingStrategy' may have shared mutable state

Any suggestions?

**Comment by adamek314** [comment]
Adding this code silenced the warning:

```
extension DatabaseUUIDEncodingStrategy: @unchecked @retroactive Sendable {
    
}
```

Although I'm still not sure why the system thinks that a static let property with an enum type is not concurrently-safe. I could understand it if it was a class instance.

**Reply by groue** [reply]
Hello @adamek314. A future GRDB version will remove those warnings. Until then, adding `@unchecked Sendable` is a correct workaround.

**Reply by groue** [reply]
Fixed in 3043d7dc, shipped in v6.28.0
